tests/php/src/Internal/DependencyManagement/AbstractServiceProviderTests.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * AbstractServiceProviderTests class file.
+ *
+ * @package Automattic\WooCommerce\Tests\Internal\DependencyManagement
+ */
+
+namespace Automattic\WooCommerce\Tests\Internal\DependencyManagement;
+
+use Automattic\WooCommerce\Internal\DependencyManagement\AbstractServiceProvider;
+use Automattic\WooCommerce\Internal\DependencyManagement\ExtendedContainer;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\ClassWithConstructorArgumentWithoutTypeHint;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\ClassWithDependencies;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\ClassWithPrivateConstructor;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\DependencyClass;
+use League\Container\Definition\DefinitionInterface;
+
+/**
+ * Tests for AbstractServiceProvider.
+ */
+class AbstractServiceProviderTests extends \WC_Unit_Test_Case {
+
+	/**
+	 * The system under test.
+	 *
+	 * @var AbstractServiceProvider
+	 */
+	private $sut;
+
+	/**
+	 * The container used for tests.
+	 *
+	 * @var ExtendedContainer
+	 */
+	private $container;
+
+	/**
+	 * Runs before each test.
+	 */
+	public function setUp() {
+		$this->container = new ExtendedContainer();
+
+		$this->sut = new class() extends AbstractServiceProvider {
```

Comment:
I use this as "system under test" is a very correct term for anything to test, and [it's an industry standard](https://en.wikipedia.org/wiki/System_under_test). But I agree that it may be unclear the first time you see it, that's why I declare the variable as follows:

```php
/**
 * The system under test.
 *
 * @var AbstractServiceProvider
 */
private $sut;
```